### PR TITLE
fix: Improve performance of global dimension search on druid

### LINKED ISSUE #4983: Global dimension search: performance at Druid scale
Follow-on to https://github.com/rilldata/rill/pull/4905. The feature should feel good for large-scale Druid-backed projects.

Notes from @begelundmuller:
- For Druid, we need to route to this API (which is not available in its SQL interface): https://druid.apache.org/docs/latest/querying/searchquery/
- Add a new runtime API for search, named something like `MetricsViewSearch`

### PR DESCRIPTION
**Pull request by AdityaHegde**
closes #4983

- [x] Add a new endpoint for global dimension search.
- [x] Make a native druid query for dimension search.
- [x] Use `UNION ALL` query for non druid olap.
- [x] Apply security filter to druid native queries. We are running `EXPLAIN PLAN` to get a json and pass on the filter object. Right now this will not always work, especially if the security filter is a subquery. We fallback to `UNION ALL` in this case.

Will do a follow up for making the requests in serial.

### CODE REVIEW

File: runtime/queries/metricsview_search.go [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+	for _, c := range inst.Connectors {
+		if c.Name == "druid" {
+			dsn = c.Config["dsn"]
+		}
+	}
+	if dsn == "" {
+		return false, fmt.Errorf("druid connector config not found in instance")
+	}
+
+	nq := druid.NewNativeQuery(strings.Replace(dsn, "/v2/sql/avatica-protobuf/", "/v2/", 1))
```
[the remark this answers is not in the document]
Comment (pjain1) [reply runtime/queries/metricsview_search.go:227]: also the suffix is `/v2/sql`

[the remark this answers is not in the document]
Comment (pjain1) [reply runtime/queries/metricsview_search.go]: yeah my mistake, for native queries its just `/v2`

**Comment by ericpgreen2** [comment]
Frontend code looks reasonable. Though the requests in my test are taking 30s+ to resolve. I'll send a JAM privately.

One frontend performance improvement – when closing the Search box, it'd be good to cancel in-flight requests.

File: runtime/queries/metricsview_search.go [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+	}
+
+	return nil
+}
+
+func (q *MetricsViewSearch) Export(ctx context.Context, rt *runtime.Runtime, instanceID string, w io.Writer, opts *runtime.ExportOptions) error {
+	return nil
+}
+
+var druidDNSFixer = regexp.MustCompile(`/v2/sql(?:/avatica-protobuf)?/?`)
```
Comment (pjain1) [inline runtime/queries/metricsview_search.go:162]: Should be `DSN` not `DNS`

nit: `avatica-protobuf` should not be present after getting dsn from druid driver. Also we can call this var `druidSqlDSN`

File: runtime/queries/metricsview_search.go [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+
+func (q *MetricsViewSearch) executeSearchInDruid(ctx context.Context, rt *runtime.Runtime, instanceID, table string, policy *runtime.ResolvedMetricsViewSecurity) (bool, error) {
+	var query map[string]interface{}
+	if policy != nil && policy.RowFilter != "" {
+		druidolap, release, err := rt.OLAP(ctx, instanceID, "druid")
```
Comment (pjain1) [inline runtime/queries/metricsview_search.go:167]: nit: `druidolap` can be just `olap`

**Review by pjain1 — APPROVED** [review]
just minor nits

**Pull request merged**